router_test.go

```diff
@@ -306,3 +307,27 @@ func TestProcedureSort(t *testing.T) {
 		},
 	}, ps, "should order procedures lexicographically on (service, procedure, encoding)")
 }
+
+func TestUnknownServiceName(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	m := NewMapRouter("service1")
+	foo := transporttest.NewMockUnaryHandler(mockCtrl)
+
+	m.Register([]transport.Procedure{
+		{
+			Name:        "foo",
+			HandlerSpec: transport.NewUnaryHandlerSpec(foo),
+			Service:     "service2",
+			Encoding:    "json",
+		},
+	})
+
+	_, err := m.Choose(context.Background(), &transport.Request{
+		Service:   "wrongService",
+		Procedure: "foo",
+		Encoding:  "json",
+	})
+	assert.Contains(t, err.Error(), `unrecognized service name "wrongService", available services: "service1" "service2"`)
```

Comment:
This test will need one minor adjustment, however. Since the iteration order of a map is not guaranteed this test may will fail some times. 

We'll need to assert that the error contains this message _or_ the same message with the service names swapped

Reply:
FYI, I believe that this is why your current build is failing

Reply:
resolved, we will sort the service names to output consistent error message.